ClosedXML/Excel/XLWorksheet.cs

```diff
@@ -1599,13 +1599,15 @@ private void CheckRangeNotInTable(XLRange range)
 
         private string GetNewTableName(string baseName)
         {
+            var existingTableNames = new HashSet<String>(this.Workbook.Worksheets.SelectMany(ws => ws.Tables).Select(t => t.Name));
+
             var i = 1;
             string tableName;
             do
             {
                 tableName = baseName + i.ToString();
                 i++;
-            } while (Tables.Any(t => t.Name == tableName));
+            } while (existingTableNames.Contains(tableName, StringComparer.OrdinalIgnoreCase));
```

Comment:
![image](https://user-images.githubusercontent.com/19576939/79971073-e5047900-84a4-11ea-8447-4bfd50b74e95.png)

Wrong `Contains`, sorry. `HashSet` cannot optionate comparer in `Contains` by design as it has to compute hashcode on adding elements. That's why I proposed to pass a comparer in the constructor. In that case, we call `.Contains(tableName)` and the searching is case-insensitive.

Reply:
Oooh, I see. Is it Resharper that gives you that warning?

Reply:
In fact, that was just me suspicious about how is it possible to pass comparer in a hashset. But once you've said that I've noticed that there was a warning from Resharper indeed which I missed 😆 

![image](https://user-images.githubusercontent.com/19576939/79972626-4fb6b400-84a7-11ea-9f56-547994913355.png)